gradle/build.go

```diff
@@ -117,3 +149,42 @@ func (b Build) Build(context libcnb.BuildContext) (libcnb.BuildResult, error) {
 
 	return result, nil
 }
+
+func handleGradleSettings(binding libcnb.Binding, gradleHome string, md map[string]interface{}) error {
+	path, ok := binding.SecretFilePath("gradle.properties")
+	if !ok {
+		return nil
+	}
+
+	err := os.MkdirAll(gradleHome, 0755)
+	if err != nil {
+		return fmt.Errorf("unable to make gradle home\n%w", err)
+	}
+
+	gradlePropertiesPath := filepath.Join(gradleHome, "gradle.properties")
+	if err := os.Symlink(path, gradlePropertiesPath); os.IsExist(err) {
+		err = os.Remove(gradlePropertiesPath)
+		if err != nil {
+			return fmt.Errorf("unable to remove old symlink for gradle.properties\n%w", err)
+		}
+
+		err = os.Symlink(path, gradlePropertiesPath)
```

Comment:
I suspect it doesn't work to link this in before `libbs` contributes the cache layer? `libbs`will replace gradle home with a link to the cached layer.

https://github.com/paketo-buildpacks/libbs/blob/main/cache.go#L44

This is the point I was making about the right level of deduplication. I am not sure the `libbs` Application and Cache constructs give quite enough flexibility to handle new cases as they arise without constantly change the library interface.

Reply:
Let me test it again and see. I thought it was working. I will double-check. I see what you're saying about the timing of when the links are created.

Reply:
Yes, so this was broken but not obviously broken. The gradle.properties file was being picked up but it was causing the cache to not be set up correctly. The symlink for the cache directory would fail thinking it already exists. I have an idea how to delay the creation of the gradle.properties symlink until after the cache is created. Working on that now.

Reply:
This should be working on the latest commit now. I added a new layer contributor between the cache and gradle. So the cache runs first, adds its symlink, then the new contributor runs adds a symlink, and finally gradle runs.

I tested the following scenarios, which all appear to be working:

1. The first run initializes the cache, picks up the bound gradle.properties file, installs gradlew and runs gradle.
2. The second run does nothing. The layer is cached as expected.
3. The third run touched the gradle.properties file. This triggers gradle to run again, but the cache layer is restored and the build is just for gradle to recompile.
4. The fourth run, the app source code is modified. This triggers gradle to run again, but the cache layer is restored and the build is just for gradle to recompile.